### ISSUE TITLE
Gdrive configuration

### ISSUE THREAD

**Issue by Chnaps** [marcussacana/DirectPackageInstaller#11]
Hi Marcus, thanks a lot for this amazing app. could you please explain a bit more on how to configure Gdrive (especially for authenticated links ?) I managed to make every other File Hosts work expect for this one. Thanks in advance ! :)

**Comment by marcussacana** [comment]
You should use this extension to extract your cookies in gdrive.
[Edge](https://microsoftedge.microsoft.com/addons/detail/get-cookiestxt/helleheikohejgehaknifdkcfcmceeip#:~:text=This%20extension%20can%20be%20super%20useful%20in%20circumstances,in%20the%20Edge%20context%20menu%20on%20any%20website.) [Chrome](https://chrome.google.com/webstore/detail/get-cookiestxt/bgaddhkoddajcdgocldbbfleckgcbcid#:~:text=This%20extension%20can%20be%20super%20useful%20in%20circumstances,menu%20on%20any%20website.%20Read%20more.%20Read%20less.) [Firefox](https://addons.mozilla.org/en-US/firefox/addon/cookies-txt/)
Basically, install the extension, open the google drive in your browser,
then Click in the extension icon and click in the "Export" Button, copy the text of the txt that the has downloaded, paste in the DirectPackageInstaller window after try open a GDrive file, then will works.

**Comment by Chnaps** [comment]
Thanks a lot for your prompt answer ! I'm going to try and decompose what I am doing so you can tell me what I am doing wrong.

1. This is my personal drive which is not public, **so I can only access it when I am authenticated** 

![image](https://user-images.githubusercontent.com/41871844/162470233-790315ec-4dab-48c2-81fe-39aec6fc570e.png)

2. I have specifically opened one pkg link for this test, and clicked the extension and export 

![image](https://user-images.githubusercontent.com/41871844/162471114-64734b46-4699-4fde-80e0-26c7f04657a5.png)

3. which gave me this file :

![image](https://user-images.githubusercontent.com/41871844/162471089-04a69a29-40ec-47d4-b068-b95d3d6ac902.png)

I can't seem to find any link in here that I can paste into DirectPackageInstaller ?

thanks again for you work and your help !

**Comment by marcussacana** [comment]
Usually you a public file url and paste in the DirectPackageInstaller, then he will ask for the cookies automatically,
But you can do manually as well, ensure your cookie files is named cookies.txt and put in the DirectPackageInstaller, then, restart it.
If not works then we have a bug here.

**Comment by marcussacana** [comment]
Also, cookies are private, it's better remove that Screenshot since can give access to your account.

**Comment by Chnaps** [comment]
I'm still getting "Failed to Open the PKG"

**Comment by marcussacana** [comment]
Shoud be a bug then, I didn't tested with private files before anyway.
That said I'm busy creating a major update to the tool, then should take a time to this update be ready.

**Comment by Chnaps** [comment]
Here's one link (public) to one of the smaller pkgs of the GDrive : 

https://drive.google.com/file/d/1JCswtwirKXA_jF46WGXt8c81nwU_J8im/view?usp=sharing

The link works, but I get "Failed to Open the PKG" in DirectPackageInstaller. Ccan you test it on your side ?

**Comment by marcussacana** [comment]
yup, buggy, isn't working.

**Comment by marcussacana** [comment]
Let'me check if is a minor bug, then I can do a hotfix now

**Comment by marcussacana** [comment]
Test it in the lastest version, the 5.2.0 version.

**Comment by Chnaps** [comment]
will test it immediately thanks !!!

**Comment by Chnaps** [comment]
you are an amazing genius, it works perfectly

**Comment by marcussacana** [comment]
Good, feel free to contact if you had any problem again

**Comment by Mc-muffin** [comment]
So, I'm not the original poster, but I ran into a similar problem, the first time trying to download a GDrive link goes "fine" (Proxied download, NO segmented download) but then the download fails on the PS4 after around 5 minutes, after that trying to add any GDrive links to the app just won't work until hours later, any ideas about that?

**Comment by marcussacana** [comment]
@Mc-muffin you have put your gdrive cookies in the DirectPackageInstaller?

**Comment by Mc-muffin** [comment]
> @Mc-muffin you have put your gdrive cookies in the DirectPackageInstaller?

Yes, I used the cookies.txt extension, and the first time ever it did show the pkg in the app and everything, but the download failed on PS4 after around 5min, now the cookies don't seem to work and show the "failed to load PKG" in the app. Only "weird" thing is that I have more than one account logged-in , but that shouldn't matter as the file i oubk

**Comment by marcussacana** [comment]
@Mc-muffin Try delete the cookies.txt in the DirectPackageInstaller folder and setup again.
Also, try enable the Proxy Mode if you didn't enabled yet, since that will be able to automatically refresh the download url if expired, while the PS4 standalone download can't do that.

**Comment by Mc-muffin** [comment]
Okay, didn't have time for testing before, but after some tests it works just the first time, after that the file (in jdownloader) shows "download quota exceeded" which is weird because the file is on a persona google drive, I guess it's because I pick the cookies while loged to 7 accounts? Does that make sense?

**Comment by marcussacana** [comment]
The login is to bypass that download quota a bit more, but still have a limitation.
The drive don't allow you download your file many times even if you own it, you should have this problem more easy if you used the Segmented Download mode, since that will open multiple connections and basically start the download multiple times, this make the gdrive limit smaller than the usual, if you got trouble with that you should disable the Segmented Download mode.
